### PR TITLE
Feature/studio wonder

### PR DESCRIPTION
**Pull request by fightwithcrayon**
Adds Wonder component to Studio page

### CODE REVIEW

File: src/templates/Studio/Wonder/Sun/Sun.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+const Sun = ({ layout, onAddSun, scene, world }) => {
+  useEffect(() => {
+    let interval;
+    let light;
+    let prevProgress = 0;
+
+    const getMinutesSinceMidnight = (timestamp = Date.now()) => {
+      const date = new Date(timestamp);
+      const utcDate = date.getTime() + (date.getTimezoneOffset() * 60000);
+      const amsterdamDate = new Date(utcDate + (3600000 * 1));
```
Comment (larixk) [inline src/templates/Studio/Wonder/Sun/Sun.js]: ` * 1` could go. Also there might be cleaner ways to get the local time, but this looks simple enough.

File: src/templates/Studio/Wonder/Wonder.js [hunk 2]
```diff
@@ -1,207 +1,109 @@
...
+      setCanvasVisible(false);
+    };
+  }, [camera, currentEngine, currentScene, world]);
 
   useEffect(() => {
     let engine;
+    let scene;
 
     const createEngine = () => {
+      // canvasRef.current.style.height = `${introRef.current.getBoundingClientRect().top + (window.pageYOffset || document.documentElement.scrollTop)}px`;
```
Comment (larixk) [inline src/templates/Studio/Wonder/Wonder.js]: Comment can go?

File: src/templates/Studio/Wonder/Wonder.module.scss [hunk 3]
```diff
@@ -5,20 +5,13 @@
...
   width: 100vw;
   z-index: 0;
-  //pointer-events: none;
-  opacity: 0;
-  transition: opacity 4000ms ease-in;
+  pointer-events: none;
+  opacity: 0.01;
+  transition: opacity 4000ms ease-in 500ms;
+  outline: none;
+  // border: 1px solid black;
```
Comment (larixk) [inline src/templates/Studio/Wonder/Wonder.module.scss]: Comment can go?

File: src/templates/Studio/Wonder/Camera/Camera.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
+import React, { useEffect } from 'react';
+import { ArcRotateCamera } from 'babylonjs';
+
+const Camera = ({ canvasRef, layout, onCreateCamera, scene }) => {
+  useEffect(() => {
+    let camera;
+    const canvas = canvasRef.current;
+
+    if (scene) {
```
Comment (larixk) [inline src/templates/Studio/Wonder/Camera/Camera.js]: Might be cleaner with early return?

File: src/templates/Studio/Wonder/Lighting/Lighting.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import {
+  HemisphericLight,
+  Vector3,
+  Color3,
+} from 'babylonjs';
+
+const Lighting = ({ scene }) => {
+  useEffect(() => {
+    let generalLight;
+    if (scene) {
```
Comment (larixk) [inline src/templates/Studio/Wonder/Lighting/Lighting.js]: Might be cleaner with early return?

File: src/templates/Studio/Wonder/Mirrors/Mirrors.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        ));
+        mirrorMaterial.reflectionTexture.renderList = reflected;
+
+        mirror.material = mirrorMaterial;
+        activeMirrorContent.push(mirror);
+        activeMirrorContent.push(mirrorMaterial);
+      });
+    };
+
+    if (scene && world && layout && layout.length > 0) {
```
Comment (larixk) [inline src/templates/Studio/Wonder/Mirrors/Mirrors.js]: Might be cleaner as early return?

File: src/templates/Studio/Studio.js [hunk 7]
```diff
@@ -55,17 +56,22 @@ export const query = graphql`
...
-      <InfoBlock key={index} collection={collection} />
-    ))}
-    <Impression data={studioPage.frontmatter.studioImpression} />
-    <Footer {...indexPage.frontmatter} />
-  </Layout>
-);
+export default ({ location, data: { indexPage, studioPage } }) => {
+  const introRef = useRef();
+  return (
+    <Layout globalClass="studio">
```
Comment (larixk) [inline src/templates/Studio/Studio.js]: Not sure what the `"studio"` globalClass actually does. Is it somewhere in css? Maybe I just missed it.

File: src/templates/Studio/Wonder/Shadows/Shadows.js [hunk 8]
```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import { ShadowGenerator } from 'babylonjs';
+
+const Shadows = ({ scene, sun, world }) => {
+  useEffect(() => {
+    let shadows;
+    if (scene && sun && world) {
```
[the remark this answers is not in the document]
Comment (fightwithcrayon) [reply src/templates/Studio/Wonder/Shadows/Shadows.js]: I'm not sure you can exit early that way within a hook – the return is meant to be the cleanup function, executed on change of a dependency. I may be wrong about that

[the remark this answers is not in the document]
Comment (larixk) [reply src/templates/Studio/Wonder/Shadows/Shadows.js]: the clean up function can change between different situations though. If there is no `scene || sun || world` you would not have to clean up the shadows either so the clean up function can be a noop or even `undefined`.

**Pull request merged**